github_backup/github_backup.py

```diff
@@ -502,12 +525,20 @@ def get_github_repo_url(args, repository):
 
     auth = get_auth(args, encode=False, for_git_cli=True)
     if auth:
-        repo_url = "https://{0}@{1}/{2}/{3}.git".format(
-            auth,
-            get_github_host(args),
-            repository["owner"]["login"],
-            repository["name"],
-        )
+        if args.token_fine is None:
+            repo_url = "https://{0}@{1}/{2}/{3}.git".format(
+                auth,
+                get_github_host(args),
+                repository["owner"]["login"],
+                repository["name"],
+            )
+        else:
+            repo_url = "https://{0}@{1}/{2}/{3}.git".format(
+                "oauth2:" + auth,
```

Comment:
Why not just return the auth with `oauth2:` on the prefix if its not a fine-grained token?

Reply:
Agreed. I will review the `if` and `else` token scenarios for simplification.

Reply:
This has now been attempted resolved.